### PR TITLE
Conda support

### PR DESCRIPTION
**Pull request by LysandreJik**
Will push a new version on anaconda cloud every time a tag starting with `v` is pushed (like `v1.2.2`).

Will appear here: https://anaconda.org/huggingface/datasets

Depends on `conda-forge` for now, so the following is required for installation:

```
conda install -c huggingface -c conda-forge datasets
```

### CODE REVIEW

**Comment by lhoestq** [comment]
Nice thanks :) 
Note that in `datasets` the tags are simply the version without the `v`. For example `1.2.1`.

**Comment by LysandreJik** [comment]
Do you push tags only for versions?

**Comment by lhoestq** [comment]
Yes I've always used tags only for versions

File: .github/workflows/release-conda.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: Release - Conda
 on:
   push:
     tags:
-      - v*
+      - '*'
```
Comment (LysandreJik) [inline .github/workflows/release-conda.yml]: Let's try it like this then! Should trigger on all tags.

Comment (lhoestq) [reply .github/workflows/release-conda.yml]: What about `[0-9]+.[0-9]+.[0-9]+*` ?

Comment (LysandreJik) [reply .github/workflows/release-conda.yml]: Sure, just pushed it!

**Review by lhoestq — APPROVED** [review]
Thank you :)

**Pull request merged**